pkg/evaluator/evaluator.go

```diff
@@ -669,13 +706,20 @@ func (e *Evaluator) evalSliceExpr(expr *parser.SliceExpression) (Value, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	var val Value
 	switch left := left.(type) {
 	case *Array:
-		return left.Slice(start, end)
+		val, err = left.Slice(start, end)
 	case *String:
-		return left.Slice(start, end)
+		val, err = left.Slice(start, end)
+	default:
+		err = fmt.Errorf("%w: expected map before '.', found  %v", ErrType, left)
```

Comment:
This error string was wrong before. May as well fix it now:
1. double space after "found"
2. should be `'['` not `'.'`

Reply:
and should probable read: expected `string` or `array` before `[` ?